### PR TITLE
Module 5: React Tests

### PR DESCRIPTION
**Pull request by Tsurhanau**

### CODE REVIEW

File: src/components/Courses/components/CourseCard/tests/CourseCard.test.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	card: mockedCoursesList[0],
+	openCardInfo: jest.fn(),
+};
+
+describe('CourseCard Component', () => {
+	beforeEach(() => {
+		useSelectorMock.mockReturnValue(mockedState.user);
+
+		render(
+			<Provider store={mockedStore}>
```
Comment (Papialoushka) [inline src/components/Courses/components/CourseCard/tests/CourseCard.test.js:31]: [OPTIONAL] It is a good idea to create a test utility that returns a wrapper for test components

File: src/components/Courses/tests/Courses.test.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+		const buttonElement = screen.getByTestId('add-button');
+
+		const spanElement = buttonElement.querySelector('span');
+
+		expect(spanElement).toHaveTextContent(BUTTON_TEXT.ADD_NEW_COURSE);
+
+		userEvent.click(buttonElement);
+
+		await act(async () => {
```
Comment (Papialoushka) [inline src/components/Courses/tests/Courses.test.js:85]: It looks like two clicks on the same button are not needed

File: src/components/Courses/tests/Courses.test.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+			fireEvent.click(buttonElement);
+		});
+
+		await waitFor(() => {
```
Comment (Papialoushka) [inline src/components/Courses/tests/Courses.test.js]: If act is awaited, await waitFor should not be needed

File: src/components/Courses/tests/Courses.test.js (hunk 2 above)
Comment (Tsurhanau) [reply src/components/Courses/tests/Courses.test.js:85]: done

File: src/components/Courses/tests/Courses.test.js (hunk 3 above)
Comment (Tsurhanau) [reply src/components/Courses/tests/Courses.test.js]: done

**Pull request merged**